sprints/sustainability/models.py

```diff
@@ -239,10 +273,43 @@ def _fetch_accounts_chunk(from_: str, to: str) -> Dict[str, Dict]:
                             account_reports.name,
                             SustainabilityAccount(account_reports.name),
                         )
-                        account.add_reports(account_reports.reports)
+                        account.add_reports(account_reports.reports, worklogs)
 
         return categories
 
+    def get_projects_sustainability(self) -> Dict[str, float]:
+        """Retrieve sustainability of all processed projects."""
+        aggregated_projects: Dict[str, Dict[str, float]] = {}
+        for category in settings.TEMPO_ACCOUNT_TRANSLATE.values():
+            for account in getattr(self, category):
+                for project, hours in account.by_project.items():
+                    aggregated_projects.setdefault(project, {})
+                    aggregated_projects[project][category] = aggregated_projects[project].get(category, 0) + hours
+
+        sustainability: Dict[str, float] = {}
+        for project, data in aggregated_projects.items():
+            billable_hours = data.get(settings.TEMPO_ACCOUNT_TRANSLATE[settings.TEMPO_BILLABLE_ACCOUNT_NAME], 0)
+            non_billable_reponsible_hours = data.get(
+                settings.TEMPO_ACCOUNT_TRANSLATE[settings.TEMPO_NON_BILLABLE_RESPONSIBLE_ACCOUNT_NAME], 0
+            )
+            cell_hours = billable_hours + non_billable_reponsible_hours
+            try:
+                responsible_hours = non_billable_reponsible_hours / cell_hours * 100
+            except ZeroDivisionError:
+                responsible_hours = float('inf')
+            sustainability[project] = responsible_hours
+
+        return sustainability
+
+    def get_accounts_remaining_time(self) -> Dict[str, float]:
+        """Retrieve time left of all fetched accounts."""
+        accounts: Dict[str, float] = {}
+        for category in settings.TEMPO_ACCOUNT_TRANSLATE.values():
+            for account in getattr(self, category):
+                accounts[account.name] = accounts.get(account.name, 0) + account.ytd_overall - account.ytd_goal
```

Comment:
This could be simplified by using a `defaultdict(float)` so just an `accounts[account.name] += ...` is necessary. Just a suggestion though, feel free to ignore.

Reply:
Good point, I did this. I couldn't do it everywhere, though, because some changes would require upfront cache invalidation for compatibility.